src/setfit/trainer_distill.py

```diff
@@ -0,0 +1,239 @@
+    ) -> None:
+        super(DistilSetFitTrainer, self).__init__(
+            model,
+            train_dataset,
+            eval_dataset,
+            model_init,
+            metric,
+            loss_class,
+            num_iterations,
+            num_epochs,
+            learning_rate,
+            batch_size,
+            seed,
+            column_mapping,
+            use_amp,
+            warmup_proportion,
+        )
+
+        self.teacher_model = teacher_model
+
+    def train(
+        self,
+        num_epochs: Optional[int] = None,
+        batch_size: Optional[int] = None,
+        learning_rate: Optional[float] = None,
+        body_learning_rate: Optional[float] = None,
+        l2_weight: Optional[float] = None,
+        trial: Union["optuna.Trial", Dict[str, Any]] = None,
+    ):
+        """
+        Main training entry point.
+
+        Args:
+            num_epochs (int, *optional*):
+                Temporary change the number of epochs to train the Sentence Transformer body/head for.
+                If ignore, will use the value given in initialization.
+            batch_size (int, *optional*):
+                Temporary change the batch size to use for contrastive training or logistic regression.
+                If ignore, will use the value given in initialization.
+            learning_rate (float, *optional*):
+                Temporary change the learning rate to use for contrastive training or SetFitModel's head in logistic regression.
+                If ignore, will use the value given in initialization.
+            body_learning_rate (float, *optional*):
+                Temporary change the learning rate to use for SetFitModel's body in logistic regression only.
+                If ignore, will be the same as `learning_rate`.
+            l2_weight (float, *optional*):
+                Temporary change the weight of L2 regularization for SetFitModel's differentiable head in logistic regression.
+            trial (`optuna.Trial` or `Dict[str, Any]`, *optional*):
+                The trial run or the hyperparameter dictionary for hyperparameter search.
+        """
+        if trial:  # Trial and model initialization
+            set_seed(self.seed)  # Seed must be set before instantiating the model when using model_init.
+            self._hp_search_setup(trial)  # sets trainer parameters and initializes model
+
+        if self.train_dataset is None:
+            raise ValueError("DistilledSetFitTrainer:: training requires a train_dataset.")
+
+        self._validate_column_mapping(self.train_dataset)
+        train_dataset = self.train_dataset
+        if self.column_mapping is not None:
+            logger.info("Applying column mapping to training dataset")
+            train_dataset = self._apply_column_mapping(self.train_dataset, self.column_mapping)
+
+        x_train = train_dataset["text"]
+        y_train = train_dataset["label"]
+        if self.loss_class is None:
```

Comment:
Should we raise a proper `ValueError` here with a message or did I misunderstand the role of this `return` statement?

Reply:
i agree, i added a ValueError with a message. BTW, we had the same issue in SetFitTrainer i also added a ValueError there.